04_numeronym/willshen8/numeronyms.go

```diff
@@ -0,0 +1,36 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"regexp"
+	"strconv"
+)
+
+var out io.Writer = os.Stdout
+
+func numeronyms(vals ...string) []string {
+	returnNumeronyms := []string{}
+	for _, val := range vals {
+		returnNumeronyms = append(returnNumeronyms, convertStringToNumeronym(val))
+	}
+	return returnNumeronyms
+}
+
+func convertStringToNumeronym(input string) string {
+	regEx := regexp.MustCompile(`[^a-zA-Z0-9]+`)
+	formattedInput := regEx.ReplaceAllString(input, "")
+
+	if len(formattedInput) <= 3 {
```

Comment:
You can manipulate the string directly. Is there a need for converting it into slice? eg: `input[0]` would return the first element of that string.

Reply:
Except for removing special characters i mean. But even that can be handled in the string itself.

Reply:
Hi, line 22 ensures that only a word is being processed. And line 23 removes spaces for processing. As "Web development" is a proper string that should be output as 'W12t' and not 'W13t' as per [https://en.wikipedia.org/wiki/Numeronym](url)

Reply:
Oh, sorry my bad. misunderstood it.